cmudb/env/docker-entrypoint.sh

```diff
@@ -0,0 +1,170 @@
+  POSTGRES_DB="noisepage"
+fi
+
+if [ -z "$POSTGRES_HOST_AUTH_METHOD" ]; then
+  POSTGRES_HOST_AUTH_METHOD="md5"
+fi
+
+if [ -z "$PGPORT" ]; then
+  PGPORT=15721
+fi
+
+# =====================================================================
+# Helper functions.
+# =====================================================================
+
+_pgctl_start() {
+  ${BIN_DIR}/pg_ctl --pgdata=${PGDATA} -w start
+}
+
+_pg_stop() {
+  ${BIN_DIR}/pg_ctl --pgdata=${PGDATA} -w stop
+}
+
+_pg_start() {
+  ${BIN_DIR}/postgres "-D" "${PGDATA}" -p 15721
+}
+
+_pg_initdb() {
+  WALDIR="--waldir=${POSTGRES_INITDB_WALDIR}"
+  if [ -z ${POSTGRES_INITDB_WALDIR} ]; then
+    WALDIR=""
+  fi
+  ${BIN_DIR}/initdb --pgdata=${PGDATA} $WALDIR ${POSTGRES_INITDB_ARGS}
+}
+
+_pg_config() {
+  AUTO_CONF=${PGDATA}/postgresql.auto.conf
+  HBA_CONF=${PGDATA}/pg_hba.conf
+
+  # pg_hba.conf
+  echo "host all all 0.0.0.0/0 ${POSTGRES_HOST_AUTH_METHOD}" >> ${HBA_CONF}
+
+  # postgresql.auto.conf
+  # Allow Docker host to connect to container.
+  echo "listen_addresses = '*'" >> ${AUTO_CONF}
+}
+
+_pg_create_user_and_db() {
+  ${BIN_DIR}/psql -c "create user ${POSTGRES_USER} with login password '${POSTGRES_PASSWORD}'" postgres
+  ${BIN_DIR}/psql -c "create database ${POSTGRES_DB} with owner = '${POSTGRES_USER}'" postgres
+}
+
+_pg_setup_replication() {
+  AUTO_CONF=${PGDATA}/postgresql.auto.conf
+  HBA_CONF=${PGDATA}/pg_hba.conf
+
+  # See PostgreSQL docs for complete description of parameters.
+
+  # wal_level: How much information to ship over.
+  echo "wal_level = replica" >> ${AUTO_CONF}
+  # hot_standby: True to enable connecting and running queries during recovery.
+  echo "hot_standby = on" >> ${AUTO_CONF}
+  # max_wal_senders: Maximum number of concurrent connections to standby/backup clients.
+  echo "max_wal_senders = 10" >> ${AUTO_CONF}
+  # max_replication_slots: Maximum number of replication slots.
+  echo "max_replication_slots = 10" >> ${AUTO_CONF}
+  # hot_standby_feedback: True if standby should tell primary about what queries are currently executing.
+  echo "hot_standby_feedback = on" >> ${AUTO_CONF}
+
+  if [ "${NP_REPLICATION_TYPE}" = "primary" ]; then
+    # ===============================
+    # Enable replication.
+    # ===============================
+
+    # Create replication user.
+    ${BIN_DIR}/psql -c "create user ${NP_REPLICATION_USER} with replication encrypted password '${NP_REPLICATION_PASSWORD}'" postgres
+    # Allow replication user to connect..
+    echo "host replication ${NP_REPLICATION_USER} 0.0.0.0/0 md5" >> ${HBA_CONF}
+    # Reload configuration.
+    ${BIN_DIR}/psql -c "select pg_reload_conf()" postgres
+    # Create replication slot for replica.
+    ${BIN_DIR}/psql -c "select pg_create_physical_replication_slot('replication_slot_replica1')" postgres
+
+    # ===============================
+    # Enable replication stats.
+    # ===============================
+    ${BIN_DIR}/psql -c "grant pg_monitor to ${POSTGRES_USER}" postgres
```

Comment:
I'm pretty sure we're going to want this on the replica too so that we can view stats there as well, do these user actions get propagated to the replica? I think this might fit better in `_pg_create_user_and_db`, that way we get stats even if replication is off.

Reply:
1. Checked with `\du`, yes, both primary and replica will have `noisepage {pg_monitor}`
2. I agree, thanks! pushed